sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/ProjectTypeCollectorTest.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * SonarSource :: .NET :: Shared library
+ * Copyright (C) 2014-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.dotnet.shared.plugins;
+
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ProjectTypeCollectorTest {
+
+  @Test
+  public void withNoProjects() {
+    ProjectTypeCollector underTest = new ProjectTypeCollector();
+    assertThat(underTest.getSummary()).isEmpty();
+  }
+
+  @Test
+  public void withNoFiles_skip() {
+    ProjectTypeCollector underTest = new ProjectTypeCollector();
+    underTest.addProjectInfo(false, false);
+
+    assertThat(underTest.getSummary()).isEmpty();
+  }
+
+  @Test
+  public void withNoFiles_twoModules_logOne() {
+    ProjectTypeCollector underTest = new ProjectTypeCollector();
+    underTest.addProjectInfo(false, false);
+    underTest.addProjectInfo(false, false);
+
+    assertThat(underTest.getSummary()).hasValue("Found 1 MSBuild projects. 1 with no MAIN or TEST files.");
```

Comment:
Why `1` when two `false, false` projects were added?

Reply:
the first is considered global module

Reply:
Can you add comment as in the other tests please? It's confusing
>// this will be skipped, considered as top-level module